File: src/lowlevel/server/icefsoperators/icefsPoll.go
/*
 * @Author: Tan90degrees tangentninetydegrees@gmail.com
 * @Date: 2023-03-11 07:18:32
 * @LastEditors: Tan90degrees tangentninetydegrees@gmail.com
 * @LastEditTime: 2023-04-18 08:03:55
 * @FilePath: /icefs/src/lowlevel/server/icefsoperators/icefsPoll.go
 * @Description:
 *
 * Copyright (C) 2023 Tan90degrees <tangentninetydegrees@gmail.com>.
 */
package icefsoperators

import (
	"context"
	pb "icefs-server/icefsgrpc"
	"icefs-server/icefsthrift"
	"syscall"
)

func (s *IcefsServer) doIcefsPoll(fakeInode uint64) (status int32, revents uint32) {
	status = int32(syscall.ENOTSUP)
	return
}

func (s *IcefsGRpcServer) DoIcefsPoll(ctx context.Context, req *pb.IcefsPollReq) (*pb.IcefsPollRes, error) {
	var res pb.IcefsPollRes

	res.Status, res.Revents = s.server.doIcefsPoll(req.Inode)

	return &res, nil
}

func (s *IcefsThriftServer) DoIcefsPoll(ctx context.Context, req *icefsthrift.IcefsPollReq) (*icefsthrift.IcefsPollRes, error) {
	var res icefsthrift.IcefsPollRes
	var revents uint32

	res.Status, revents = s.server.doIcefsPoll(uint64(req.Inode))
	res.Revents = icefsthrift.Ui32(revents)

	return &res, nil
}

File: src/lowlevel/client/icefsoperators/icefsPoll.cpp
/*
 * @Author: Tan90degrees tangentninetydegrees@gmail.com
 * @Date: 2023-03-30 04:19:29
 * @LastEditors: Tan90degrees tangentninetydegrees@gmail.com
 * @LastEditTime: 2023-04-20 10:27:54
 * @FilePath: /icefs/src/lowlevel/client/icefsoperators/icefsPoll.cpp
 * @Description:
 *
 * Copyright (C) 2023 Tan90degrees <tangentninetydegrees@gmail.com>.
 */
#include <stdio.h>

#include <string>

#include "icefsClient.hpp"
#include "icefsClientUtils.hpp"

void IcefsClient::DoIcefsPoll(fuse_req_t fuseReq, fuse_ino_t inode,
                              struct fuse_file_info *fi,
                              struct fuse_pollhandle *ph) {
  ICEFS_PR_FUNCTION;

  fuse_reply_err(fuseReq, ENOTSUP);
}